common/rpc/encryption/localStoreCertProvider.go

```
@@ -77,57 +92,142 @@ func (s *localStoreCertProvider) FetchServerCertificate() (*tls.Certificate, err
 	return s.serverCert, nil
 }
 
+func (s *localStoreCertProvider) fetchServerCertificateFromInline() (*tls.Certificate, error) {
+	if s.tlsSettings.Server.CertData == "" {
```

Shouldn't we return an error here?